File: app/src/main/java/com/duy/pascal/backend/exceptions/ChangeValueConstantException.java
package com.duy.pascal.backend.exceptions;

import com.duy.pascal.backend.linenumber.LineInfo;

/**
 * Created by Duy on 13-Apr-17.
 */

public class ChangeValueConstantException extends ParsingException {
    public String name;
    public Object value;

    public ChangeValueConstantException(LineInfo line, String message, String name) {
        super(line, message);
        this.name = name;
    }

    public ChangeValueConstantException(LineInfo line, String message, String name, Object value) {
        super(line, message);
        this.name = name;
        this.value = value;
    }

    public ChangeValueConstantException(LineInfo line) {
        super(line);
    }
}

File: libCompiler/src/main/java/com/duy/pascal/backend/exceptions/value/ChangeValueConstantException.kt
/*
 *  Copyright (c) 2017 Tran Le Duy
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.duy.pascal.backend.exceptions.value

import com.duy.pascal.backend.exceptions.ParsingException
import com.duy.pascal.backend.linenumber.LineInfo

/**
 * Created by Duy on 13-Apr-17.
 */

class ChangeValueConstantException : ParsingException {
    var name: String = ""
    var value: Any? = null

    constructor(line: LineInfo, message: String, name: String) : super(line, message) {
        this.name = name
    }

    constructor(line: LineInfo, message: String, name: String, value: Any) : super(line, message) {
        this.name = name
        this.value = value
    }

    constructor(line: LineInfo) : super(line) {}
}
